react/src/lib/components/DeckGLMap/components/Map.tsx

```diff
@@ -108,12 +111,14 @@ const Map: React.FC<MapProps> = ({
                     }}
                     ref={refCb}
                     onHover={onHover}
+                    onViewStateChange={({viewState}) => setviewState({viewState})}
                 >
                     {children}
                 </DeckGL>
                 {showInfoCard ? <InfoCard pickInfos={hoverInfo} /> : null}
                 <Settings />
-            </ReduxProvider>
+                {viewState ? <DistanceScale zoomLevel={viewState} /> : null}
```

Comment:
It's better to send just what you need to a comp. You don't need the whole view state in DistanceScale, so here you can send just `viewState.zoom`.